src/libraries/System.Security.Cryptography/tests/osslplugins/e_dntest.c

```diff
@@ -0,0 +1,171 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include <string.h>
+
+#define OPENSSL_SUPPRESS_DEPRECATED
+#include <openssl/engine.h>
+
+static const char* g_engineId = "dntest";
+static const char* g_engineName = "DotNet Test ENGINE";
+
+// PKCS#1 format
+static char g_pubRsaKey[140] = {
+    0x30,0x81,0x89,0x02,0x81,0x81,0x00,0xBF,0x67,0x16,0x84,0x85,0x21,0x5A,0x6A,0xB8,
+    0x9B,0xCA,0xB9,0x33,0x1F,0x6F,0x5F,0x36,0x0F,0x43,0x00,0xBE,0x5C,0xF2,0x82,0xF7,
+    0x70,0x42,0x95,0x7E,0xA2,0x02,0x90,0x8B,0x22,0x79,0xF3,0x4A,0x42,0x6D,0x62,0xF5,
+    0x9D,0x6C,0x10,0x56,0xE3,0x6D,0xC9,0xF6,0xEE,0xA9,0xAE,0xB1,0xB3,0x1F,0x81,0x22,
+    0xF5,0x83,0xEE,0x9C,0xAE,0x2A,0x86,0xA4,0x71,0x44,0x90,0x5D,0xF0,0x54,0x41,0xB0,
+    0xA5,0xF2,0x9E,0x03,0xC5,0xAC,0x18,0x88,0xD9,0x37,0x44,0xD8,0x96,0x38,0xD8,0x3A,
+    0xC3,0x77,0x74,0xB3,0x39,0xE4,0xAF,0xB3,0x49,0xC7,0x14,0xB1,0x22,0x38,0xB0,0xF8,
+    0x1A,0x71,0x38,0x0F,0x05,0x1C,0x58,0x5C,0xB2,0x74,0x34,0xFA,0x54,0x4B,0xDA,0xC6,
+    0x79,0xE1,0xE1,0x65,0x81,0xD0,0xE9,0x02,0x03,0x01,0x00,0x01
+};
+
+// PKCS#1 format
+char g_priRsaKey[608] = {
+    0x30,0x82,0x02,0x5C,0x02,0x01,0x00,0x02,0x81,0x81,0x00,0xBF,0x67,0x16,0x84,0x85,
```

Comment:
When I was doing development on this, the public key named "first" and the private key named "first" didn't match (not that you necessarily could have seen that since I wrote it to load keys from files that I didn't check in).  It looks like you've changed that here so that they're the same.

I thought it was valuable that they were mismatched, to show that the public/private namespace mattered.

Whether they're the same (as-is) or different (because you change it up), add a comment saying what their relationship is.

Reply:
added a comment. I intentionally called the same to ensure we don't accidentally load the wrong one when same name is used